lightning/src/ln/channelmanager.rs

```diff
@@ -1156,8 +1156,13 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			return Err(APIError::APIMisuseError { err: format!("Channel value must be at least 1000 satoshis. It was {}", channel_value_satoshis) });
 		}
 
+		let their_features = {
+			let per_peer_state = self.per_peer_state.read().unwrap();
+			let peer_state = per_peer_state.get(&their_network_key).unwrap().lock().unwrap();
```

Comment:
This is failing in nearly every test, I believe the issue is the same as the one addressed by #971.